hmt_escrow/job.py

```diff
@@ -646,23 +651,35 @@ def bulk_payout(
         func_args = [eth_addrs, hmt_amounts, url, hash_, 1]
 
         try:
-            handle_transaction_with_retry(txn_func, self.retry, *func_args, **txn_info)
-            return self._bulk_paid() is True
+            tx_receipt = handle_transaction_with_retry(
+                txn_func,
+                self.retry,
+                *func_args,
+                **txn_info,
+            )
+            bulk_paid = self._check_transfer_event(tx_receipt)
 
         except Exception as e:
             LOG.warn(
                 f"{txn_event} failed with main credentials: {self.gas_payer}, {self.gas_payer_priv} due to {e}. Using secondary ones..."
             )
 
+        if bulk_paid:
+            return bulk_paid
+
+        LOG.warn(
+            f"{txn_event} failed with main credentials: {self.gas_payer}, {self.gas_payer_priv}. Using secondary ones..."
+        )
+
         raffle_txn_res = self._raffle_txn(
```

Comment:
not in try...catch, unlike handle_transaction_with_retry

Reply:
`try...except` is inside `_raffle_txn`